src/Redis/Queues/RedisQueue.cs

```diff
@@ -176,7 +181,7 @@ public class RedisQueue<T> : QueueBase<T> where T : class {
             return String.Concat("q:", _queueName, ":in");
         }
 
-        public override async Task<string> EnqueueAsync(T data) {
+        protected override async Task<string> EnqueueImplAsync(T data) {
             string id = Guid.NewGuid().ToString("N");
             _logger.Debug("Queue {_queueName} enqueue item: {id}", _queueName, id);
```

Comment:
Would be good to move the logging into the base class! and maybe pass id to the imp method? or should we leave id generation up to the client?

@ejsmith @jamie94bc